### DISCUSSION TITLE
OPA Decision ID but no/nullable result

### DISCUSSION THREAD

**Discussion by AdrianArnautu** [open-policy-agent/community#432] [OPA and Rego]
Hi folks,

We have a OPA server configured with bundle API and from time to time we're seeing a response that has a decision ID but no result

Something like (input omitted)
```json
{"decision_id":"abc4fec1-c11a-dda1-8edd-10f31f832a91","result":null}
```
or
```json
{"decision_id":"abc4fec1-c11a-dda1-8edd-10f31f832a91","input":{"data":"here"}}
```

I need your help to understand what would be the combination of events as I can't reproduce it.
I know that if there wouldn't be data loaded in OPA we would get at least an empty map - like 
```json
{"decision_id":"xxx", "result": {}}
```

OPA v0.52.0

**Comment by anderseknert** [comment]
Hi @AdrianArnautu! It's hard to tell without seeing the policy. Is there anything in that which could cause the evaluated rule to be null? 🤔

**Reply by AdrianArnautu** [reply]
No, our policies should always return a response.
If there's a missing bundle/policy, OPA responds with `"result": {}`

Looking further into the issue, there's a timing match with the issue we described here https://github.com/open-policy-agent/opa/issues/5868
Looks like whenever there's a `fatal error: concurrent map iteration and map write` almost always there's an empty response

**Reply by anderseknert** [reply]
That seems like the likely culprit then. Thanks for raising this — I'll make sure to include it for prioritization.

**Comment by ashutosh-narkar** [comment]
@AdrianArnautu are you using the `http.send` builtin in your policy? If you could share your policy that would be helpful. Thanks.

**Reply by AdrianArnautu** [reply]
hi @ashutosh-narkar,
no, we're using the bundle plugin together with the discovery plugin (5s polling interval) 
the bundle contains one rbac policy and data with resources and actions ( ~ 300k)
plus ~ 150 custom policies using the dynamic composition checking attributes from the input  

as per metrics, the bundle takes about 1.2 - 1.5s to load into OPA

**Reply by ashutosh-narkar** [reply]
@AdrianArnautu [here's](https://github.com/open-policy-agent/opa/issues/5868#issuecomment-1581583426) one explanation for the `fatal error: concurrent map iteration and map write` error. Feel free to comment in the issue with your thoughts.
